packages/protocol-kit/src/Safe.ts

```diff
@@ -114,10 +120,38 @@ class Safe {
   private async init(config: SafeConfig): Promise<void> {
     const { provider, signer, isL1SafeSingleton, contractNetworks } = config
 
-    this.#safeProvider = new SafeProvider({
-      provider,
-      signer
-    })
+    const isPasskeySigner = signer && typeof signer !== 'string'
+
+    if (isPasskeySigner) {
+      const safeProvider = new SafeProvider({
+        provider
+      })
+      const chainId = await safeProvider.getChainId()
+      const customContracts = contractNetworks?.[chainId.toString()]
+
+      const safeWebAuthnSignerFactoryContract = await getSafeWebAuthnSignerFactoryContract({
```

Comment:
This can be retrieved from the safeProvider right?

Reply:
yep, but I updated this code to use the `createSafeProvider` util function to wrap all the logic about to create the `safeProvider` object to reuse it in all packages.